XUtils.c

```diff
@@ -337,3 +339,25 @@ ssize_t full_write(int fd, const void* buf, size_t count) {
 
    return written;
 }
+
+/* Compares floating point values for ordering data entries. In this function,
+   NaN is considered "less than" any other floating point value (regardless of
+   sign), and two NaNs are considered "equal" regardless of payload. */
+int compareRealNumbers(double a, double b) {
+   int result = (int)isgreater(a, b) - isgreater(b, a);
```

Comment:
```suggestion
   int result = (int)isgreater(a, b) - (int)isgreater(b, a);
```

Shouldn't this actually include both casts. And AFAICS both should be implicit anyway.

Reply:
The cast may be removed or both made explicit. I don't know which looks neater.

Reply:
Removed is neater.